### PR TITLE
Apply LinterProvider to DiffView

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-code-manager/issues/441

The next patches are going to be a bit big so maybe it's easier if I break them up.

### CODE REVIEW

**Comment by kumar303** [comment]
I had an afterthought about the `DiffView` testing strategy. I think it will be simpler without requiring a full store. Stay tuned...

**Comment by kumar303** [comment]
@willdurand it's ready for a look now

**Comment by willdurand** [comment]
>  The diff coverage is `90.47%`.

Do you know why the diff coverage is not 100%?

**Comment by kumar303** [comment]
It seems to be saying that the initialization of parameters to `simulateLinterProvider()` are not covered: https://codecov.io/gh/mozilla/addons-code-manager/pull/443/diff

This is odd because the function is definitely executed but only in tests (maybe it ignores tests?). When I run `yarn test --coverage` locally it says that `test-helpers.tsx` has 100% line coverage.

**Comment by kumar303** [comment]
I'm not worried about this sort of coverage because TypeScript covers it better. I'll see if I can just ignore it with a comment.

File: src/components/DiffView/index.spec.tsx [hunk 1]
```diff
@@ -62,73 +53,32 @@ describe(__filename, () => {
...
+    messagesAreLoading = false,
+    selectedMessageMap = undefined,
+    ...renderParams
+  }: RenderWithLinterProviderParams = {}): ShallowWrapper => {
+    const root = render(renderParams);
+
+    return simulateLinterProvider(root, {
+      messageMap,
+      messagesAreLoading,
+      selectedMessageMap,
```
Comment (willdurand) [inline src/components/DiffView/index.spec.tsx]: 😍

File: src/components/DiffView/index.tsx [hunk 2]
```diff
@@ -253,30 +195,19 @@ export class DiffViewBase extends React.Component<Props> {
...
+    );
+  }
+}
 
-export default withRouter<PublicProps & Partial<DefaultProps> & RouterProps>(
-  connect(mapStateToProps)(DiffViewBase),
-) as React.ComponentType<PublicProps & Partial<DefaultProps>>;
+export default withRouter(DiffViewBase) as React.ComponentType<
+  PublicProps & Partial<DefaultProps>
+>;
```
Comment (willdurand) [inline src/components/DiffView/index.tsx]: Is this still needed here? Can't we type the `withRouter` HOC instead?

```
withRouter<PublicProps & Partial<DefaultProps> & RouterProps>(DiffViewBase);
```

I believe if `PublicProps` and `DefaultProps` were the same type, we would not even have to add types to `withRouter` (it works in `CodeView`):

```
withRouter(DiffViewBase);
```

**Pull request merged**